tinygrad/runtime/ops_python.py

```diff
@@ -100,15 +103,20 @@ def __call__(self, *bufs, global_size:Tuple[int,int,int]=(1,1,1), local_size:Tup
             ul[i] = [idxs[2-arg[0]]] * warp_size
           elif arg[1][0] == 'l':
             ul[i] = [x[2-arg[0]] for x in warp]
-        elif uop is UOps.CONST: ul[i] = [int(arg) if dtypes.is_int(dtype) else float(arg)] * warp_size
+        elif uop is UOps.CONST:
+          casted_arg = int(arg) if dtypes.is_int(dtype) else float(arg)
+          if dtype.count > 1:
+            ul[i] = [[casted_arg] * warp_size for _ in range(dtype.count)]
+          else:
+            ul[i] = [casted_arg] * warp_size
         elif uop is UOps.DEFINE_ACC:
           if dtype.count > 1:
             ul[i] = [[arg] * warp_size for _ in range(dtype.count)]
           else:
             ul[i] = [arg] * warp_size
         elif uop is UOps.LOOP:
-          if i not in ul:
-            ul[i] = [0] * warp_size
+          if i not in ul or ul[i][0] == inp[1][0]:
```

Comment:
I don't understand this or

Reply:
There are two cases that you need to (re)initialize the loop for:
- The first is when you enter the loop for the first time. This is initialization because `ul` won't have any state about this loop.
- The second is when you enter the loop again after you have finished all its iterations. This only happens in nested loops. When the nested loop executes its last iteration, the loop will be entered again because of the next iteration of the upper loop, which will be problematic because the state of the previous iteration of the nested loop is still saved in `ul` and needs to be reinitialized to `inp[0][0]`. But now while explaining it, I realize that the best solution is to clear the state of the loop after the last iteration. I will push a commit with that.